osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/sfc/CheckServiceFunctionChainMetaTask.java

```diff
@@ -0,0 +1,157 @@
+import org.osc.core.broker.model.plugin.ApiFactoryService;
+import org.osc.core.broker.model.sdn.NetworkElementImpl;
+import org.osc.core.broker.service.exceptions.VmidcBrokerValidationException;
+import org.osc.core.broker.service.persistence.DeploymentSpecEntityMgr;
+import org.osc.core.broker.service.tasks.TransactionalMetaTask;
+import org.osc.sdk.controller.api.SdnRedirectionApi;
+import org.osc.sdk.controller.element.NetworkElement;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(service = CheckServiceFunctionChainMetaTask.class)
+public class CheckServiceFunctionChainMetaTask extends TransactionalMetaTask {
+	@Reference
+	CreateServiceFunctionChainTask createServiceFunctionChainTask;
+
+	@Reference
+	UpdateServiceFunctionChainTask updateServiceFunctionChainTask;
+
+	@Reference
+	DeleteServiceFunctionChainTask deleteServiceFunctionChainTask;
+
+	@Reference
+	private ApiFactoryService apiFactoryService;
+
+	private SecurityGroup securityGroup;
+	TaskGraph tg;
+	private List<NetworkElement> portPairGroups;
+
+	public CheckServiceFunctionChainMetaTask create(SecurityGroup sg) {
+		CheckServiceFunctionChainMetaTask task = new CheckServiceFunctionChainMetaTask();
+
+		task.createServiceFunctionChainTask = this.createServiceFunctionChainTask;
+		task.updateServiceFunctionChainTask = this.updateServiceFunctionChainTask;
+		task.deleteServiceFunctionChainTask = this.deleteServiceFunctionChainTask;
+		task.securityGroup = sg;
+		task.apiFactoryService = this.apiFactoryService;
+		task.dbConnectionManager = this.dbConnectionManager;
+		task.txBroadcastUtil = this.txBroadcastUtil;
+
+		return task;
+	}
+
+	@Override
+	public void executeTransaction(EntityManager em) throws Exception {
+		this.tg = new TaskGraph();
+		this.securityGroup = em.find(SecurityGroup.class, this.securityGroup.getId());
+
+		if (this.securityGroup.getNetworkElementId() == null && this.securityGroup.getServiceFunctionChain() != null) {
+
+			this.portPairGroups = getPortPairGroupWithSameProjectId(em, this.securityGroup.getProjectId(),
+					this.securityGroup.getServiceFunctionChain().getVirtualSystems());
+
+			this.tg.appendTask(this.createServiceFunctionChainTask.create(this.securityGroup, this.portPairGroups));
+
+		} else if (this.securityGroup.getNetworkElementId() != null && this.securityGroup.getServiceFunctionChain() != null) {
+
+			this.portPairGroups = getPortPairGroupWithSameProjectId(em, this.securityGroup.getProjectId(),
+					this.securityGroup.getServiceFunctionChain().getVirtualSystems());
+
+			try (SdnRedirectionApi sdnApi = this.apiFactoryService
+					.createNetworkRedirectionApi(this.securityGroup.getVirtualizationConnector())) {
+
+				List<NetworkElement> sfcPortPairGroups = sdnApi
+						.getNetworkElements(new NetworkElementImpl(this.securityGroup.getNetworkElementId()));
+
+				if (!this.portPairGroups.equals(sfcPortPairGroups)) {
```

Comment:
dont do equals as part of network element. Create a method which can take both the list and compare them with getElementId